src/Modules/Trinity.DynamicCluster/Trinity.ServiceFabric.GarphEngine.Runtime/TrinitySeverRuntimeMangerBase.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Fabric;
+using System.Fabric.Query;
+using System.IO;
+using System.Threading;
+using Trinity.Diagnostics;
+using Trinity.Network;
+using Trinity.ServiceFabric.GarphEngine.Infrastructure.Interfaces;
+
+namespace Trinity.ServiceFabric.GarphEngine.Infrastructure
+{
+    public abstract class TrinitySeverRuntimeMangerBase : ITrinityServerRuntimeManager
+    {
+        private TrinityServer m_trinityServer;
+
+        // We manage and treat this groping of data as immutable
+        private readonly (List<System.Fabric.Query.Partition> Partitions,
```

Comment:
right, because SF has a fixed number of partitions. I was planning to also support dynamic partitioning but figured that it would bring extra complexity to a lot of stuff.. so I rejected the idea.